### PR TITLE
Provide a charter for SIG Big Data

### PR DESCRIPTION
**Pull request by erikerlandson**
Adds a `charter.md` for the `sig-big-data` directory, and makes corresponding adjustments to `sigs.yaml` and the SIG readme.

### CODE REVIEW

File: sig-big-data/charter.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# SIG Big Data Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [sig-governance].
+
+## Scope
+
+The Big Data SIG serves as a community resource for advising big data and data science related software projects on techniques and best practices for integrating with Kubernetes. This SIG also represents the concerns of users from big data communities to Kubernetes for the purposes of driving new features and other enhancements, based on big data use cases.
```
Comment (timothysc) [inline sig-big-data/charter.md]: This is isn't really a definition of a SIG, b/c there isn't code ownership in k/k.  Instead, it's more of a working group.

File: sig-big-data/charter.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+### In scope
+
+#### Code, Binaries and Services
```
Comment (timothysc) [inline sig-big-data/charter.md]: Are there OWNERS files maintained by the SIG in the Kubernetes org?

**Comment by timothysc** [comment]
/assign @bgrant0607 

I know you have thoughts because you've mentioned them before.

File: sig-big-data/charter.md (hunk 2 above)
Comment (erikerlandson) [reply sig-big-data/charter.md]: currently, no

**Comment by timothysc** [comment]
In thinking this over, it really seems like this should be a sub-project of @kubernetes/sig-apps-pr-reviews

**Comment by philips** [comment]
I want to echo concerns by the other steering committee members @spiffxp and @timothysc that this doesn't seem to meet the code ownership requirements of a SIG.

**Comment by timothysc** [comment]
Per conversations, we are planning to hold on rationalizing this until post-KubeCon.

**Comment by timothysc** [comment]
It's the new year.  

@mattfarina I'm kind of in the boat that this should be a sig-apps sponsored working group.  Thoughts?

**Comment by pwittrock** [comment]
Injecting myself into the conversation...

@erikerlandson What do you need to accomplish the goals outlined in the charter?  E.g. is it just about having a mailing list, slack channel, zoom, calendar, etc.  Because SIGs are the decision making bodies for the project, they need to have clear accountability and processes for making decisions.  If the work you are doing does not need to be accountable to the Kubernetes project (e.g. release, docs, arch, etc), we should probably come up with a different designation.

**Comment by pwittrock** [comment]
> The thing that still puzzles me about the WG designation is that it implies some task (research a problem, draft a design, report back, etc), with a finite lifetime. However we might categorize our group's activities, I view them as on-going.

This is a valid point.  I think the accountability aspect of a SIG designation is a bigger issue than the finite aspect of a WG designation.  We've kicked around the idea of BirdsOfAFeather (BOF) groups that are similar to WG's without requiring any affiliation with SIGs and still allowing them to be on-going.

**Comment by pwittrock** [comment]
@erikerlandson Sounds like there is rough agreement on the needs.  BoF type groups were a topic of discussion at the SC meeting today.  I believe "User Group" was a name folks seemed to like.

There is a rough plan to create a "User Group" (final name TBD) designation that provides collaboration artifacts such as:

- mailing list
- slack channel
- zoom
- calendar
- community page (e.g. a directory for the UG with a README.md)

A question we will want to nail down is if / how should UGs be associated with SIGs.  e.g. if everyone stops participating in a UG, who is responsible for cleaning it up?  I don't think this would have much impact on day-to-day operations of the UG.

**Comment by liyinan926** [comment]
@pwittrock what are the major differences between a UG (as being planned) and a WG?

**Comment by pwittrock** [comment]
@liyinan926 

Working Groups are designed to facilitate targeted work across SIGs.  As the are currently defined they are not targeted at facilitating support or discussion for an area.  UGs would be targeted at this case.

From a practical standpoint I believe you would get the same resources, but may require a different organizational structure.  I imagine the requirements for starting a WG would be higher than a UG - e.g. the SIGs accountable for the work need to be well defined and the vision needs to be shared by participating SIGs.

**Comment by erikerlandson** [comment]
@pwittrock I agree, the plan around User Groups sounds like a good alignment with our goals. FWIW, I would suggest that a UG need not be affiliated with a SIG. If a UG becomes inactive, the co-chairs should be responsible for some kind of official halting of activities. Perhaps there should be a contingency for unresponsive co-chairs but that seems like a corner case.

**Comment by pwittrock** [comment]
@erikerlandson ACK.  We also need to work through what it is practical to provide to UGs without moderation.

**Comment by timothysc** [comment]
Sounds like we are in agreement here.  @erikerlandson did you want to close this PR, or wait till we formalize user groups?

**Comment by erikerlandson** [comment]
@timothysc I'll leave this open for now, and close it when we are in a position to re-submit it as a User Group

**Comment by bgrant0607** [comment]
FYI, user groups now officially exist as a concept:

https://github.com/kubernetes/community/blob/master/governance.md#community-groups

**Comment by erikerlandson** [comment]
@bgrant0607 IIUC it says user groups are to be documented in sigs.yaml - is the idea that there will be a new `usergroups:` section there?

**Comment by timothysc** [comment]
@kubernetes/sig-contributor-experience-bugs  ^

**Comment by timothysc** [comment]
I think we can close this issue now and follow up with other PRs and issues to cross-reference.  Contributor experience can shake out the logistics on the .yaml deets.

**Comment by nikhita** [comment]
>  Contributor experience can shake out the logistics on the .yaml deets.

I have created https://github.com/kubernetes/community/pull/3298 to add support for User Groups in sigs.yaml.

**Comment by erikerlandson** [comment]
Thank you @nikhita!

**Comment by erikerlandson** [comment]
Work on this has been moved to #3304

**Pull request closed without merging**